lib/ansible/module_utils/oneview.py

```diff
@@ -0,0 +1,782 @@
+
+            result = self.execute_module()
+
+            if "changed" not in result:
+                result['changed'] = False
+
+            self.module.exit_json(**result)
+
+        except HPOneViewException as exception:
+            self.module.fail_json(msg='; '.join(str(e) for e in exception.args))
+
+    def resource_absent(self, resource, method='delete'):
+        """
+        Generic implementation of the absent state for the OneView resources.
+        It checks if the resource needs to be removed.
+        Args:
+            resource (dict): Resource to delete.
+            method (str):
+                Function of the OneView client that will be called for resource deletion. Usually delete or remove.
+
+        Returns:
+            A dictionary with the expected arguments for the AnsibleModule.exit_json
+
+        """
+        if resource:
+            getattr(self.resource_client, method)(resource)
+
+            return {"changed": True, "msg": self.MSG_DELETED}
+        else:
+            return {"changed": False, "msg": self.MSG_ALREADY_ABSENT}
+
+    def get_by_name(self, name):
+        """
+        Generic get by name implementation.
+        Args:
+            name: Resource name to search for.
+
+        Returns:
+            The resource found or None.
+        """
+        result = self.resource_client.get_by('name', name)
+        return result[0] if result else None
+
+    def resource_present(self, resource, fact_name, create_method='create'):
+        """
+        Generic implementation of the present state for the OneView resources.
+        It checks if the resource needs to be created or updated.
+
+        Args:
+            resource (dict):
+                Resource to create or update.
+            fact_name (str):
+                Name of the fact returned to the Ansible.
+            create_method (str):
+                Function of the OneView client that will be called for resource creation. Usually create or add.
+
+        Returns:
+            A dictionary with the expected arguments for the AnsibleModule.exit_json
+        """
+
+        changed = False
+        if "newName" in self.data:
+            self.data["name"] = self.data.pop("newName")
+
+        if not resource:
+            resource = getattr(self.resource_client, create_method)(self.data)
+            msg = self.MSG_CREATED
+            changed = True
+
+        else:
+            merged_data = resource.copy()
+            merged_data.update(self.data)
+
+            if ResourceComparator.compare(resource, merged_data):
+                msg = self.MSG_ALREADY_PRESENT
+            else:
+                resource = self.resource_client.update(merged_data)
+                changed = True
+                msg = self.MSG_UPDATED
+
+        return dict(
+            msg=msg,
+            changed=changed,
+            ansible_facts={fact_name: resource}
+        )
+
+    @staticmethod
+    def transform_list_to_dict(list_):
```

Comment:
Generally, staticmethod is a good indicator that this should be a function at the toplevel of the module_util, not a method of the class.  This looks like a prime example of that.